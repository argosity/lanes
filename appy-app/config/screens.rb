
Lanes::Screen.define "base" do | screen |
    screen.description = "Base Screen for Appy App"
    screen.icon        = ""
    screen.group_id    = ""
    screen.model_class = ""
    screen.view_class  = "AppyApp.Screens.Base"
end
