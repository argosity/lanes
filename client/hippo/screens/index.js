import { observable, computed } from 'mobx';
import { map, filter, intersection } from 'lodash';
import Group from './group';
import Config from '../config';

const Screens = observable({

    all: observable.map(),

    get activeGroups() {
        const gids = Group.enabled_group_ids || map(Group.all, 'id');
        const groups = map(gids, gid => Group.forId(gid));
        return filter(groups, group => intersection(group.screens, this.active).length);
    },

    @computed get active() {
        return map(Config.screen_ids, id => this.all.get(id));
    },

    reset() {
        Config.screen_ids.clear();
    },

});

export default Screens;
