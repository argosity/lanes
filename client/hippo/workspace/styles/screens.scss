.screen {
    padding: 5px;
    width: 100%;

    > .first-run {
        font-size: 1.3rem;
        li + li {
            margin-top: 1rem;
        }
        .help {
            margin-top: 2rem;
            p { text-align: right; }
        }
    }
}
