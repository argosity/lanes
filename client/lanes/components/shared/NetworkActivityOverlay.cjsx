class Lanes.Components.NetworkActivityOverlay extends Lanes.React.Component

    propTypes:
        model:   Lanes.PropTypes.Model.isRequired
        message: React.PropTypes.string
        timeout: React.PropTypes.number
        visible: React.PropTypes.bool
        errorTimeout: React.PropTypes.number

    getDefaultProps: ->
        timeout: 30000, errorTimeout: 2000

    removeMessage: ->
        return unless @isMounted()
        @setState(isRequesting: false, hasError: false)

    clearTimeout: ->
        clearTimeout(@state.removeHandler) if @state.removeHandler

    installRemoval: (state) ->
        @clearTimeout()
        @setState(removeHandler: _.delay(@removeMessage,
            if state.hasError then @props.errorTimeout else @props.timeout
        ))

    listenNetworkEvents: true

    setDataState: (state) ->
        if state.hasError or state.isRequesting
            @installRemoval(state)
        else if not @state.hasError
            @removeMessage()
        @setState(state)

    render: ->
        return null unless @props.visible or @state.isRequesting or @state.hasError
        message = @props.message or (
            if @state.hasError
                errorMsg = @model.errorMessage
                if _.isString(errorMsg) then errorMsg else "Error"
            else if @props.visible or @state.isRequesting is 'GET'
                'Loading…'
            else
                'Saving…'
        )
        icon = if @state.hasError then 'exclamation-circle' else 'spinner'
        <div className="overlay">
            <div className="mask" />
            <div className="message">
                <LC.Icon type={icon} animated={not @state.hasError} />
                {message}
            </div>
        </div>
