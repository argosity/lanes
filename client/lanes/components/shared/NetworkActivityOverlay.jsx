import React from 'react';

//#= require 'lanes/components/shared/Icon'

export class NetworkActivityOverlay extends React.Component {
    static propTypes = {
        model:   Lanes.PropTypes.Model,
        message: React.PropTypes.string,
        timeout: React.PropTypes.number,
        visible: React.PropTypes.bool,
        errorTimeout: React.PropTypes.number
    };

    static listenNetworkEvents = true;
    getDefaultProps() {
        return (
            {timeout: 30000, errorTimeout: 2000}
        );
    }

    removeMessage() {
        if (!this.isMounted()) { return; }
        return (
            this.setState({isRequesting: false, hasError: false})
        );
    }

    clearTimeout() {
        if (this.state.removeHandler) { return clearTimeout(this.state.removeHandler); }
    }

    installRemoval(state) {
        this.clearTimeout();
        return (
            this.setState({removeHandler: _.delay(this.removeMessage,
                state.hasError ? this.props.errorTimeout : this.props.timeout
            )})
        );
    }

    setNetworkActivity(state) {
        if (state.hasError || state.isRequesting) {
            this.installRemoval(state);
        } else if (!this.state.hasError) {
            this.removeMessage();
        }
        return (
            this.setState(state)
        );
    }

    render() {
        let errorMsg;
        if (!this.props.visible && !this.state.isRequesting && !this.state.hasError) { return null; }
        const message = this.props.message || (
            this.state.hasError ?
                (errorMsg = this.model.errorMessage,
                errorMsg && _.isString(errorMsg) ? errorMsg : "Error")
            : this.state.isRequesting === 'GET' ?
                'Loading…'
            : _.includes(['PATCH', 'POST', 'PUT'], this.state.isRequesting) ?
                'Saving…'
            : this.state.isRequesting === 'DELETE' ?
                'Deleting…'
            :
                'Pending…'
        );
        const icon = this.state.hasError ? 'exclamation-circle' : 'spinner';
        const export classes = _.classnames('overlay', {rounded: this.props.roundedCorners});
        return (
            React.createElement("div", {"export className": (classes)},
                React.createElement("div", {"export className": "mask"}),
                React.createElement("div", {"export className": "message"},
                    React.createElement(LC.Icon, {"type": (icon), "animated": (!this.state.hasError)}),
                    (message)
                )
            )
        );
    }
}
