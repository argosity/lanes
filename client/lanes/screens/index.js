import { observable, autorun } from 'mobx';
import { map, uniq } from 'lodash';
import Group from './group';
import Config from '../config';
import Sync from '../models/sync';

import user from '../user';

const Screens = observable({

    all: observable.map(),

    active: observable.array(),

    groups: observable.map(),

    get activeGroups() {
        return uniq(map(this.active, s => Group.forId(s.group_id)));
    },

    refresh() {
        Sync.perform(`${Config.api_path}/lanes/screens`).then(({ data: { screens: screenIds } }) => {
            this.active.replace(map(screenIds, id => this.all.get(id)));
        });
    },

});
export default Screens;

let previousLoggedIn = null;
autorun(() => {
    if (user.isLoggedIn !== previousLoggedIn){
        previousLoggedIn = user.isLoggedIn;
        Screens.refresh();
    }
});
