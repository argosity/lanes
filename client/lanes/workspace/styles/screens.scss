.screen {
    padding: 5px;
    //  overflow: hidden;
    width: 100%;
    .toolbar {
        margin-top: 10px;
    }

    .toolbar{
        @extend .navbar;
        @extend .navbar-default;

        .btn {
            @extend .btn-default;
            @extend .navbar-btn;
        }

        .arrow {
            border-style: solid;

            border-left-color: transparent;
            border-right-color: transparent;

            border-bottom-color: $popover-title-bg;
            border-top-color: $popover-title-bg !important; 

            border-width: 0 9px 9px;
            height: 0;
            left: 7px;
            margin-top: 10px;
            opacity: 1;
            position: absolute;
            top: -18px;
            transition: all 0.25s ease 0s;
            width: 0;
            z-index: 10;
        }
        .navbar-left {
            float: left;
        }
        .navbar-right{
            float: right;
            .arrow {
                left: unset;
                right: 10px;
            }
        }

        .dropdown-toggle {
            .badge{
                padding: 2px 2px 0;
                position: relative;
                font-size: 70%;
                top: -6px;
                right: 5px;
            }
            position: relative;
            padding-right: 2px;
            i {
                /* position: absolute; */
                /* left: 0px; */
            }
        }

    }

}
