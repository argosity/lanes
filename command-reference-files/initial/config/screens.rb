# An example of how to define a screen group
# Hippo::Screen.define_group 'appy-app-screens' do | group |
#     group.title       = "AppyApp Screens"
#     group.description = "Screens relating to AppyApp"
#     group.icon        = "heart"
# end

Hippo::Screen.for_extension 'appy-app' do | screens |

end
