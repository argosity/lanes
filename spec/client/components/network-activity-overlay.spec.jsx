import React from 'react';

import NetworkActivityOverlay from 'hippo/components/network-activity-overlay';
import { Box } from '../test-models';
import { Snapshot } from 'hippo/testing/screens';

jest.useFakeTimers();

describe('Network Activity Overlay', () => {
    let model;
    beforeEach(() => {
        model = new Box();
    });

    it('renders', () => {
        const activity = mount(<NetworkActivityOverlay model={model} />);
        expect(activity).not.toHaveRendered('.mask');
        model.syncInProgress = { method: 'PUT' };
        expect(activity).toHaveRendered('.mask');
        expect(activity.find('span').text()).toEqual('Savingâ€¦');
        jest.runAllTimers();
        expect(activity).not.toHaveRendered('.mask');
    });

    it('matches snapshot', () => {
        model.syncInProgress = { method: 'DELETE' };
        jest.runAllTimers();
        expect(Snapshot(<NetworkActivityOverlay model={model} />)).toMatchSnapshot();
    });
});
