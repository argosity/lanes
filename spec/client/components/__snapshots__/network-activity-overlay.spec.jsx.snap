// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Network Activity Overlay matches snapshot 1`] = `
.c2 {
  height: 48px;
  width: auto;
  -webkit-animation: iVXCSc 0.7s linear infinite;
  animation: iVXCSc 0.7s linear infinite;
}

.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  position: absolute;
  z-index: 11;
  height: 100%;
  width: 100%;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 span {
  margin-left: 1rem;
}

.c1 .mask {
  background-color: white;
  top: 0;
  height: 100%;
  left: 0%;
  width: 100%;
  opacity: 0.8;
  position: absolute;
}

.c1.rounded .mask {
  border-radius: 8px;
}

.c1 .message {
  min-height: 65px;
  line-height: 3rem;
  border-width: 2px;
  border-color: #865CD6;
  border-style: solid;
  margin-left: auto;
  margin-right: auto;
  font-size: 2rem;
  position: relative;
  top: 25%;
  border-radius: 8px;
  padding: 10px;
  box-shadow: 6px 7px 5px #49516F;
  background-color: background-color:#865CD6;
  color: #FFFFFF;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 80%;
}

<div
  className="c0"
>
  <div
    className="network-activity-overlay c1"
  >
    <div
      className="mask"
    >
      <div
        className="message"
      >
        <svg
          className="c2"
          size="large"
          viewBox="0 0 40 40"
          x="0px"
          y="0px"
        >
          <path
            d="M20.201,5.169c-8.254,0-14.946,6.692-14.946,14.946c0,8.255,6.692,14.946,14.946,14.946 s14.946-6.691,14.946-14.946C35.146,11.861,28.455,5.169,20.201,5.169z M20.201,31.749c-6.425,0-11.634-5.208-11.634-11.634 c0-6.425,5.209-11.634,11.634-11.634c6.425,0,11.633,5.209,11.633,11.634C31.834,26.541,26.626,31.749,20.201,31.749z"
            fill="#000"
            opacity="0.2"
          />
          <path
            className="dark"
            d="M26.013,10.047l1.654-2.866c-2.198-1.272-4.743-2.012-7.466-2.012h0v3.312h0 C22.32,8.481,24.301,9.057,26.013,10.047z"
            fill="#000"
          />
        </svg>
        <span>
          Deletingâ€¦
        </span>
      </div>
    </div>
  </div>
</div>
`;
